prboom2/src/s_advsound.c

```diff
@@ -59,16 +59,17 @@ musinfo_t musinfo;
 //
 void S_ParseMusInfo(const char *mapid)
 {
-  memset(&musinfo, 0, sizeof(musinfo));
-  musinfo.current_item = -1;
-
-  S_music[NUMMUSIC].lumpnum = -1;
 
   if (W_CheckNumForName("MUSINFO") != -1)
   {
     int num, lumpnum;
     int inMap = false;
 
+    memset(&musinfo, 0, sizeof(musinfo));
```

Comment:
Hm, so this only resets the musinfo state if a MUSINFO lump is available. I am afraid that this breaks if you have a map with both UMAPINFO defined music and MUSINFO defined music changers on the map. But it's just a feeling...

Reply:
Hmm, this is a fair point, because UMAPINFO re-uses the `MUSINFO` lump functions to change music, if they are both present, the "music restarting" UMAPINFO bug will reappear. But I am not sure if we can properly fix this without disassociating the two.

Let me take another look and see if perhaps we can keep both...